src/generic/toolbkg.cpp

```diff
@@ -302,28 +305,42 @@ bool wxToolbook::InsertPage(size_t n,
     m_maxBitmapSize.x = wxMax(bitmap.GetWidth(), m_maxBitmapSize.x);
     m_maxBitmapSize.y = wxMax(bitmap.GetHeight(), m_maxBitmapSize.y);
 
+    int toolId = page->GetId();
     GetToolBar()->SetToolBitmapSize(m_maxBitmapSize);
-    GetToolBar()->AddRadioTool(n + 1, text, bitmap, bitmap.ConvertToDisabled(), text);
+    GetToolBar()->InsertTool(n, toolId, text, bitmap, bitmap.ConvertToDisabled(), wxITEM_RADIO);
 
-    if (bSelect)
+    // fix current selection
+    if (m_selection < 0)
     {
-        GetToolBar()->ToggleTool(n, true);
+        DoShowPage(page, true);
         m_selection = n;
     }
-    else
-        page->Hide();
+    else if (m_selection >= (int) n)
+    {
+        DoShowPage(page, false);
+        m_selection += 1;
+    }
+    else {
+        DoShowPage(page, false);
+    }
+
+    if (bSelect)
```

Comment:
Another minor style thing:
```suggestion
    if ( bSelect )
```